__tests__/utils/permission.test.js

```diff
@@ -0,0 +1,102 @@
+import StringConstant from '../../src/utils/string/StringConstant';
+
+jest.mock('react-native-permissions', () => {
+  return mock;
+});
+
+describe('Util permission should correct', () => {
+  beforeAll(() => {
+    jest.mock('react-native/Libraries/Utilities/Platform', () => ({
+      OS: 'android', // or 'ios'
+      select: () => null
+    }));
+  });
+  it('checkCameraPermission granted should run correctly', async () => {
+    jest.spyOn(mock, 'check').mockResolvedValue('granted');
+    const checkPermission = await checkCameraPermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionGranted);
+  });
+  it('checkCameraPermission unavailable should run correctly', async () => {
+    jest.spyOn(mock, 'check').mockResolvedValue('unavailable');
+    const checkPermission = await checkCameraPermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionUnavailable);
+  });
+  it('checkCameraPermission denied should run correctly', async () => {
+    jest.spyOn(mock, 'check').mockResolvedValue('denied');
+    const checkPermission = await checkCameraPermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionDenied);
+  });
+
+  it('checkCameraPermission default should run correctly', async () => {
+    jest.spyOn(mock, 'check').mockResolvedValue('default');
+    const checkPermission = await checkCameraPermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual('');
+  });
+});
+
+describe('requestExternalStoragePermission shpuld run correctly', () => {
+  beforeAll(() => {
+    jest.mock('react-native/Libraries/Utilities/Platform', () => ({
+      OS: 'android', // or 'ios'
+      select: () => null
+    }));
+  });
+  it('requestExternalStoragePermission granted should run correctly', async () => {
+    jest.spyOn(mock, 'check').mockResolvedValue('granted');
+    const checkPermission = await requestExternalStoragePermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionGranted);
+  });
+  it('requestExternalStoragePermission unavailable should run correctly', async () => {
+    jest.spyOn(mock, 'check').mockResolvedValue('unavailable');
+    const checkPermission = await requestExternalStoragePermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionUnavailable);
+  });
+  it('requestExternalStoragePermission denied should run correctly', async () => {
+    jest.spyOn(mock, 'check').mockResolvedValue('denied');
+    const checkPermission = await requestExternalStoragePermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionGranted);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In this test case, when the external storage permission is denied, the expected message should be `StringConstant.cameraPermissionDenied` instead of `StringConstant.cameraPermissionGranted`.

```diff
-70:     expect(checkPermission.message).toEqual(StringConstant.cameraPermissionGranted);
+70:     expect(checkPermission.message).toEqual(StringConstant.cameraPermissionDenied);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->